### PR TITLE
test: add tests for components/editor

### LINKED ISSUE #15199: [CAL-3805] Add tests for Editor
More information: [https://github.com/calcom/cal.com/pull/15198](https://github.com/calcom/cal.com/pull/15198)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3805](https://linear.app/calcom/issue/CAL-3805/add-tests-for-editor)</sub>

### PR DESCRIPTION
**Pull request by ImBIOS**
## What does this PR do?

- Fixes #15199
- Fixes CAL-3805
- Add tests for the Editor's dependencies:
  - nodes/VariableNode
  - plugins/AddVariablesDropdown
  - plugins/AutoLinklPlugin
  - plugins/ToolbarPlugin

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set? Nothing modified.
- What are the minimal test data to have? Don't understand what it means.
- What is expected (happy path) to have (input and output)? Don't understand what it means.
- Any other important info that could help to test that PR:
```bash
yarn vitest packages/ui/components/editor --coverage.enabled --coverage.include packages/ui/components/editor
```

NOTE: The above command requires me to install `@vitest/coverage-v8@^0.34.6` which matches the current `vitest` version. I'm not committing the `package.json` and `yarn.lock` changes because, in my opinion, it will make the PR Review for these test files more complicated.

```sh
 Test Files  6 passed (6)
      Tests  50 passed (50)
   Start at  03:53:36
   Duration  3.18s (transform 689ms, setup 880ms, collect 3.78s, tests 817ms, environment 1.79s, prepare 452ms)

 % Coverage report from v8
---------------------------|---------|----------|---------|---------|----------------------------------------------------------------------------------------------------------------------------------
File                       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                                                                                                
---------------------------|---------|----------|---------|---------|----------------------------------------------------------------------------------------------------------------------------------
All files                  |   70.13 |    83.03 |   78.57 |   70.13 |                                                                                                                                  
 editor                    |      98 |    92.85 |      50 |      98 |                                                                                                                                  
  Editor.tsx               |   97.56 |    92.85 |      50 |   97.56 | 50-51,100                                                                                                                        
  ExampleTheme.ts          |     100 |      100 |     100 |     100 |                                                                                                                                  
  index.ts                 |     100 |      100 |     100 |     100 |                                                                                                                                  
 editor/nodes              |     100 |      100 |     100 |     100 |                                                                                                                                  
  VariableNode.ts          |     100 |      100 |     100 |     100 |                                                                                                                                  
 editor/plugins            |   62.72 |    77.77 |   65.21 |   62.72 |                                                                                                                                  
  AddVariablesDropdown.tsx |     100 |      100 |     100 |     100 |                                                                                                                                  
  AddVariablesPlugin.tsx   |   57.86 |    71.42 |      80 |   57.86 | 29-44,53-55,61-62,100-110,121-155                                                                                                
  AutoLinkPlugin.tsx       |     100 |      100 |     100 |     100 |                                                                                                                                  
  ToolbarPlugin.tsx        |   57.03 |    75.43 |      50 |   57.03 | 50-60,62-202,212-217,229-238,253-262,265-270,273-278,283-284,286-287,289-290,295-296,309-311,323,331-340,377-382,411-413,423-424 
---------------------------|---------|----------|---------|---------|----------------------------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by ImBIOS** [comment]
What filename casing should we use for test files? I just copy/paste the file I want to test and add `.test` in the middle of it.

**Review by Amit91848 — APPROVED** [review]
LGTM! Thanks @ImBIOS

Will wait for a final review from @CarinaWolli

**Comment by Amit91848** [comment]
> What filename casing should we use for test files? I just copy/paste the file I want to test and add `.test` in the middle of it.

We do it the same way. That's fine

**Pull request merged**
